### PR TITLE
feat: add azure-identity go example

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

**Requirements**

- [ ] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

part of #187 

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [ ] no

**Notes for Reviewers**:

### CODE REVIEW

File: examples/azure-identity/go/Makefile [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+## Images
+## --------------------------------------
+
+# Output type of docker buildx build
+OUTPUT_TYPE ?= type=registry
+
+ALL_OS = linux windows
+ALL_ARCH.linux = amd64 arm64
+ALL_ARCH.windows = amd64
+ALL_OSVERSIONS.windows := 1809 1903 1909 2004 ltsc2022
```
Comment (sozercan) [inline examples/azure-identity/go/Makefile:17]:
```suggestion
ALL_OSVERSIONS.windows := 1809 ltsc2022
```

i think rest are eol?

File: examples/azure-identity/go/Makefile [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ALL_OS_ARCH.linux = $(foreach arch, ${ALL_ARCH.linux}, linux-$(arch))
+ALL_OS_ARCH.windows = $(foreach osver, ${ALL_OSVERSIONS.windows}, windows-$(osver)-$(foreach arch, ${ALL_ARCH.windows},$(arch)))
+ALL_OS_ARCH = $(foreach os, $(ALL_OS), ${ALL_OS_ARCH.${os}})
+
+# The architecture of the image
+ARCH ?= amd64
+# OS Version for the Windows images: 1809, 1903, 1909, 2004, ltsc2022
```
Comment (sozercan) [inline examples/azure-identity/go/Makefile:24]:
```suggestion
# OS Version for the Windows images: 1809, ltsc2022
```

**Review by sozercan — APPROVED** [review]
LGTM

File: examples/azure-identity/go/windows.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
+ARG BUILDER=mcr.microsoft.com/oss/go/microsoft/golang:1.19-bullseye
+ARG SERVERCORE_CACHE=gcr.io/k8s-staging-e2e-test-images/windows-servercore-cache:1.0-linux-amd64-${OS_VERSION:-1809}
```
Comment (sozercan) [inline examples/azure-identity/go/windows.Dockerfile:2]: do we need this?

File: examples/azure-identity/go/Makefile (hunk 1 above)
Comment (aramase) [reply examples/azure-identity/go/Makefile:17]: Confirmed those versions are EOL:
- https://learn.microsoft.com/en-us/lifecycle/announcements/windows-10-1903-end-of-servicing
- https://learn.microsoft.com/en-us/lifecycle/announcements/windows-10-1909-end-of-servicing
- https://learn.microsoft.com/en-us/lifecycle/announcements/windows-10-version-2004-end-of-servicing

File: examples/azure-identity/go/windows.Dockerfile (hunk 3 above)
Comment (aramase) [reply examples/azure-identity/go/windows.Dockerfile:2]: yeah, that is required for the `netapi32.dll`. The network calls fail without it.

**Pull request merged**